pkg/ssh/connection_native.go

```diff
@@ -180,3 +181,56 @@ func nativeConnectionScp(options ConnectionScpOptions) (*ConnectionScpReport, er
 
 	return &ConnectionScpReport{Response: remotePath}, nil
 }
+
+func nativeConnectionDial(options ConnectionDialOptions) (*ConnectionDialReport, error) {
```

Comment:
Nit: nativeDial seems like a sufficient name.

Reply:
I'd like to keep with the naming convention of the other functions unless we want to change all of those too